sources/core/Stride.Core.Reflection/TypeDescriptors/SetDescriptor.cs

```diff
@@ -199,33 +169,17 @@ public override object GetValue(object set, int index)
         /// <param name="set">The set.</param>
         /// <param name="index">Index of value.</param>
         /// <returns></returns>
-        public override void SetValue(object set, object index, object value)
+        public override void SetValue(object set, object index, object? value)
         {
-            if (set == null) throw new ArgumentNullException(nameof(set));
-            var simpleSet = set as ISet<object>;
-            if (simpleSet != null)
+            ArgumentNullException.ThrowIfNull(set);
+
+            if (Contains(set, index))
             {
-                if (simpleSet.Contains(index))
-                {
-                    simpleSet.Remove(index);
-                }
-                if (!simpleSet.Contains(value))
-                {
-                    simpleSet.Add(value);
-                }
+                Remove(set, index);
             }
-            else if (containsMethod != null && addMethod != null && removeMethod != null)
+            if(!Contains(set, index))
             {
-                object[] indexParam = new[] { index };
-                object[] valueParam = new[] { value };
-                if ((bool)containsMethod.Invoke(set, indexParam))
-                {
-                    removeMethod.Invoke(set, indexParam);
-                }
-                if (!(bool)containsMethod.Invoke(set, valueParam))
-                {
-                    addMethod.Invoke(set, valueParam);
-                }
+                Add(set, index);
```

Comment:
Is using `{ set, index }` instead of `{ set, value }` a bug on the previous version or a typo in your changes ? Or am I not understanding this function ?

Reply:
if (simpleSet != null)
is always true as IsSet is true when System.Collection.Generic.ISet is provided
there is no System.Collection.ISet and all the descriptors work like "on instantiation check if type fits and after that trust me"

so the else in the original code is never invoked ever

Reply:
Yep, I get that part, what I was referring to was this section from the previous logic:
```cs
if (simpleSet.Contains(index))
    simpleSet.Remove(index);
if (!simpleSet.Contains(value))
    simpleSet.Add(value);
```
Which, from my understanding, directly maps to your:
```cs
if (Contains(set, index))
    Remove(set, index);
if(!Contains(set, index))
    Add(set, index);
```

Notice the difference between the versions of the second `if` test, the previous version passes the `value` parameter (`!simpleSet.Contains(value)`) while your version uses the `index` parameter `!Contains(set, index)`
Is that intended ?

Reply:
[8fc1f8b](https://github.com/stride3d/stride/pull/2266/commits/8fc1f8bb5d3534b47e9043106213a6839d0d240f)
made it back to what it was before